source/smart_pointer/smart_pointer.hpp

```diff
@@ -2,16 +2,55 @@
 #ifndef SOURCE_SMART_POINTER_SMART_POINTER_HPP_
 #define SOURCE_SMART_POINTER_SMART_POINTER_HPP_
 #include <cstddef>
+#include <iostream>
+#include <ostream>
 
 /*
  * Smart Pointer que implementa un tipo de estrategia por conteo de referencias.
  * Permite que varios SmartPointers puedan acceder al mismo recurso compartido.
  * Cuando el ˙ltimo puntero es eliminado, se elimina y libera el recurso.
  */
+class PtrCounter {
+public:
+    PtrCounter()
+        : m_counter(0){};
+ 
+    PtrCounter(const PtrCounter&) = delete;
+    PtrCounter& operator=(const PtrCounter&) = delete;
+    ~PtrCounter()
+    {
+    }
+ 
+    void reset() {
+        m_counter = 0;
+    }
+ 
+    unsigned int get() {
+        return m_counter;
+    }
+    void operator++() {
+        m_counter++;
+    }
+ 
+    void operator++(int) {
+        m_counter++;
+    }
+    void operator--() {
+        m_counter--;
+    }
+    void operator--(int) {
+        m_counter--;
+    }
+ 
+private:
+    unsigned int m_counter{};
```

Comment:
Puedes usar el inicializador por defecto disponible a partir de c++11

```suggestion
    unsigned int m_counter = 0;
```